src/cascade/input_data/configuration/construct_country.py

```diff
@@ -217,3 +226,187 @@ def convert_gbd_ids_to_dismod_values(with_ids_df, age_groups_df):
     return dropped.rename(columns={"age_group_years_start": "age_lower", "age_group_years_end": "age_upper"})
+
+
+def compute_covariate_age_time_dimensions(covariates):
+    """
+    Determines if the input covariate data is by_age and/or by_time.
+
+    Returns:
+        dictionary: keys indicate if age or time are points (0d) vs. 1d
+    """
+
+    covar_at_dims = {}
+
+    covar_at_dims["age_1d"] = len(set(zip(covariates["age_lower"], covariates["age_upper"]))) > 1
+
+    covar_at_dims["time_1d"] = len(set(zip(covariates["time_lower"], covariates["time_upper"]))) > 1
+
+    return covar_at_dims
+
+
+def compute_covariate_age_interval(covariates):
+    """
+    Create an interval expressing all the ages in the covariates data frame.
+    If the covariates have age ranges like: [5,10], [10,15], [30,35], [35-80],
+    the overall age interval would be: [5,15], [30,80]
+    This allows checks like: 9.5 in age_interval (yes), 20.25 in age_interval (no),
+    0.667 in age_interval (no), 95 in age_interval (no), 50.1 in age_interval (yes)
+
+    Returns:
+        Interval: of ages in the covariates data frame
+    """
+    age_interval = it.empty()
+
+    age_groups = set(zip(covariates["age_lower"], covariates["age_upper"]))
+
+    for age in age_groups:
+        age_interval = age_interval | it.closed(age[0], age[1])
+
+    return age_interval
+
+
+def assign_interpolated_covariate_values(measurements, sex, covariates):
+    """
+    Compute a column of covariate values to assign to the measurements.
+
+    Args:
+        measurements (pd.DataFrame):
+            Columns include ``age_lower``, ``age_upper``, ``time_lower``,
+            ``time_upper``. All others are ignored.
+        sex (pd.Series): The sex covariate as [-0.5, 0, 0.5], corresponding to the measurements.
+        covariates (pd.DataFrame):
+            Columns include ``age_lower``, ``age_upper``, ``time_lower``,
+            ``time_upper``, ``sex``, and ``value``.
+    Returns:
+        pd.Series: One row for every row in the measurements.
+    """
+
+    # is the covariate by_age, does it have multiple years?
+    covar_at_dims = compute_covariate_age_time_dimensions(covariates)
+
+    # identify the overall interval for the covariate ages, could have middle gaps
+    covar_age_interval = compute_covariate_age_interval(covariates)
+
+    # find a matching covariate value for each measurement
+    covariate_column = compute_interpolated_covariate_values_by_sex(
+        measurements, covariates, covar_at_dims, covar_age_interval)
+
+    return covariate_column
+
+
+def get_measurement_data_by_sex(measurements):
+    """Split the measurement data by sex values found in the measurements.
+    Args:
+        measurements (pandas.DataFrame): data for a specific measurement
+    Returns:
+        dict: possible sex keys (-0.5, 0, 0.5) and measurement data as values
+    """
+    measurements_by_sex = {}
+
+    for sex in (FEMALE, MALE, BOTH):
```

Comment:
NEITHER?

Reply:
I added NEITHER, and treated it like BOTH.